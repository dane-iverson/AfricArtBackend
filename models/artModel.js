import mongoose from "mongoose";

const artSchema = mongoose.Schema(
  {
    // Title of the artwork
    title: {
      type: String,
      required: true,
    },
    // Name of the artist
    artist: {
      type: String,
      required: true,
    },
    // Size of the artwork
    size: {
      type: String,
      required: true,
    },
    // Medium used for the artwork (e.g., oil on canvas)
    medium: {
      type: String,
      required: true,
    },
    // Description of the artist
    description: {
      type: String,
      required: true,
      maxLength: 500, // Limit description to 500 characters
    },
    // URL to the image of the artwork
    image: {
      type: String,
      required: true,
    },
  },
  {
    // Automatically add timestamps for creation and updates
    timestamps: true,
  }
);

// Create and export the Art model based on the schema
export const Art = mongoose.model("Art", artSchema);
