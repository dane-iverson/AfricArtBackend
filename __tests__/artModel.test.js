import mongoose from "mongoose";
import { Art } from "../models/artModel.js";
import { MongoMemoryServer } from "mongodb-memory-server";

let mongod;

beforeAll(async () => {
  mongod = await MongoMemoryServer.create();
  const uri = mongod.getUri();
  await mongoose.connect(uri, {
    useNewUrlParser: true,
    useUnifiedTopology: true,
  });
});

afterAll(async () => {
  await mongoose.connection.close();
  await mongod.stop();
});

test("should create an Art document and match snapshot", async () => {
  const artData = {
    title: "Artwork Title",
    artist: "Artist Name",
    size: "Large",
    medium: "Oil on Canvas",
    description: "A beautiful piece of art.",
    image: "image_url",
  };

  const art = await Art.create(artData);
  expect(art.toObject()).toMatchSnapshot();
});
