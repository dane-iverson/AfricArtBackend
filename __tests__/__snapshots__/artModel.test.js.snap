// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create an Art document and match snapshot 1`] = `
{
  "__v": 0,
  "artist": "Artist Name",
  "description": "A beautiful piece of art.",
  "image": "image_url",
  "medium": "Oil on Canvas",
  "size": "Large",
  "title": "Artwork Title",
}
`;
